conformance/utils/tls/tls.go

```diff
@@ -37,6 +37,31 @@ func MakeTLSRequestAndExpectEventuallyConsistentResponse(t *testing.T, r roundtr
 	req := http.MakeRequest(t, &expected, gwAddr, "HTTPS", "https")
 
 	WaitForConsistentTLSResponse(t, r, req, expected, timeoutConfig.RequiredConsecutiveSuccesses, timeoutConfig.MaxTimeToConsistency, cPem, keyPem, server)
+
+}
+
+// MakeTLSRequestAndExpectFailure makes a request with the given parameters,
+// expecting that the request will fail but match the expectedResponse.
+func MakeTLSRequestAndExpectFailure(t *testing.T, r roundtripper.RoundTripper, gwAddr string, cPem, keyPem []byte, server string, expected http.ExpectedResponse) {
+	t.Helper()
+
+	req := http.MakeRequest(t, &expected, gwAddr, "HTTPS", "https")
+
+	req.KeyPem = keyPem
+	req.CertPem = cPem
+	req.Server = server
+
+	cReq, cRes, err := r.CaptureRoundTrip(req)
+	if err != nil {
+		tlog.Logf(t, "Request failed, not ready yet: %v", err.Error())
+		return
+	}
+
+	if err := http.CompareRequest(t, &req, cReq, cRes, expected); err != nil {
```

Comment:
This part is common between this function and `MakeTLSRequestAndExpectEventuallyConsistentResponse`. It would be better if we modularize this into a seperate function to reduce the redundancy in the code.

Reply:
I'm not going to change this because the error messages aren't the same for the single call vs repeated call.  The one enclosed in `AwaitConvergence` (repeated call) includes the elapsed time, which is helpful information.